xds/pkg/server/server.go

```diff
@@ -33,40 +38,99 @@ const (
 	grpcMaxConcurrentStreams = 1000000
 )
 
+// HealthCheck implements the health check logic exposed by the server.
+// It is implemented by internal components of the server and returns
+// an error if the component is unhealthy.
+type HealthCheck func(ctx context.Context) error
+
 // Server is a wrapper around go-control-plane's xds server.
 type Server struct {
 	logger *log.Logger
-	// port is the port the server runs on.
-	port int16
+	// xdsPort is the port the XDS server runs on.
+	xdsPort int16
 	// snapshotCache is passed updated by us and read by go-control-plane
 	// to provided proxies with config from.
 	snapshotCache cache.SnapshotCache
 	// nodeIDCh is nilable. It is used as a callback mechanism from go-control-plane's server to
 	// let us know when a new proxy has connected. The proxy's ID is passed on the channel.
 	nodeIDCh chan<- string
+	// healthChecks contains the server's health check implementations.
+	healthChecks []HealthCheck
+	// adminPort is the port the admin server runs on.
+	adminPort int16
 }
 
 // NewServer returns a new Server
 func New(
 	logger *log.Logger,
-	port int16,
+	xdsPort int16,
```

Comment:
This should fail linting 🤔 , since the function name doesn't match the comment (and it should be `NewServer`)

Reply:
not sure how to configure it, golint fails as expected but golangci-lint doesn't do this check

Reply:
Here's the config we use for Agones:
https://github.com/googleforgames/agones/blob/97dbf4677ac1c5223e434bca8ffc84654f6ae851/.golangci.yml#L47-L65

There's lots more linters 😃 (want to do it here, or in aseparatee PR?)

Reply:
TIL ([source](https://go.dev/blog/package-names#:~:text=When%20a%20function%20returns%20a%20value%20of%20type%20pkg.T%2C%20where%20T%20is%20not%20Pkg%2C%20the%20function%20name%20may%20include%20T%20to%20make%20client%20code%20easier%20to%20understand.%20A%20common%20situation%20is%20a%20package%20with%20multiple%20New%2Dlike%20functions%3A))

> When a function returns a value of type pkg.T, where T is not Pkg, the function name may include T to make client code easier to understand. A common situation is a package with multiple New-like functions:

So since this returns a Server, `New` is perfectly acceptable as a constructor name!

Reply:
Updated to use the same linters, (though none of those still catches the functionname/comment matching)

Reply:
Weird. 🤔  Seems fine now though.